addons/Treasury/Treasury.lua

```diff
@@ -103,7 +104,15 @@ function force_check()
     end
 
     -- Check inventory for unwanted items
-    if settings.AutoDrop then
+    if areas.Cities:contains(res.zones[windower.ffxi.get_info().zone].name) then
+        if settings.AutoDrop and settings.CityDrop then
+            for index, item in pairs(items.inventory) do
+                if type(item) == 'table' and code.drop:contains(item.id) then
+                    drop(item.id, index, item.count)
+                end
+            end
+        end
+    elseif settings.AutoDrop then
```

Comment:
This entire block you added can be merged with the previous block if you just change the condition:
```lua
    if settings.AutoDrop and (settings.CityDrop or not areas.Cities:contains(res.zones[windower.ffxi.get_info().zone].name)) then
```